lib/exprengine.cpp

```diff
@@ -2669,109 +2676,148 @@ static std::string execute(const Token *start, const Token *end, Data &data)
         }
 
         if (Token::Match(tok, "for|while (") && Token::simpleMatch(tok->linkAt(1), ") {")) {
+            const Token *cond = tok->next()->astOperand2();
+            const ExprEngine::ValuePtr condValue = executeExpression(cond, data);
```

Comment:
hmm.. for loops needs some tricks.

    for (int i = 0 ; i < 10; ++i)

1. the `cond` token should be the `<` token.
2. before `cond` is executed, I believe we want to execute `i = 0`.
3. I guess the `canBeFalse` should be false here.

This for would require some different handling:

    for (x : y)

Reply:
I agree, but let's make it out of scope of this PR.

I found some problems with current `for` implementation. For now we always execute `int i = 0` in the condition statement, and it stills alive, after the loop ends.

Reply:
this code is also used as a fallback for for loops.